subsys/net/l2/openthread/openthread.c

```diff
@@ -386,8 +390,10 @@ int openthread_send(struct net_if *iface, struct net_pkt *pkt)
 
 	net_capture_pkt(iface, pkt);
 
-	if (notify_new_tx_frame(pkt) != 0) {
-		net_pkt_unref(pkt);
+    if (!IS_ENABLED(CONFIG_OPENTHREAD_HDLC_RCP)) {
```

Comment:
Actually, some more details about why the changes in this file are needed would be helpful, currently I'm a bit confused. How does the transfer work if we compile out those lines? Plus it seems it'd leak a packet then?

Reply:
Hi @rlubos,
The code dedicated for the buffers management is not located on the host layer. It is done in the RCP radio controller.
It is why this part is not necessary with RCP design.